core/domain/question_domain.py

```diff
@@ -794,6 +794,13 @@ def partial_validate(self):
                 'of strings, received %s'
                 % self.inapplicable_skill_misconception_ids)
 
+        if not (all(elem.count('-') == 1 for elem in (
+                self.inapplicable_skill_misconception_ids))):
```

Comment:
Should you check that the second part is a number and that the first part has a certain length (or has the valid format for a skill ID)?

Reply:
Done. Added the same for tagged_skill_misconception_id field in AnswerGroup model.

Reply:
DO we need to audit the new check validation check for `tagged_skill_misconception_id` introduced in the exploration through this PR?

Reply:
> I'm not sure what's the length of demo skill-ids (the length of demo exploration is 1 or 2 but the general exp_id length is 12). Also, the strict id length checks seem like model-specific validation check* (Note: You may have to update lots of the test skills as all the dummy skills we create are of variable length)

Hi @DubeySandeep, the 12 char limit is not really model specific. It seems to be applicable to models other than Skills too (see models that use base_model.get_new_id). So I think the current approach of having this as a regex and using it where required makes sense.

Yep, there are lots of tests where the `tagged_skill_misconception_id` is not following this format. I will fix them in my next commit,


> DO we need to audit the new validation check for `tagged_skill_misconception_id` introduced in the exploration through this PR?

I expect this field to already follow the pattern because we have not changed the logic for generating skill ids since this field was introduced. Do you think it would be worth it to check anyway?
/cc @aks681